pkg/ast/spl/spl.peg

```diff
@@ -2030,6 +2036,107 @@ Subseconds <- ("us" / "ms" / "cs" / "ds") {
     return timeUnit, nil
 }
 
+// Returns *structs.QueryAggregators
+TransactionBlock <- PIPE CMD_TRANSACTION txnOptions:TransactionOptions {
+    queryAgg := &structs.QueryAggregators{
+        PipeCommandType: structs.TransactionType,
+        TransactionArguments: txnOptions.(*structs.TransactionArguments),
+    }
+    return queryAgg, nil
+}
+
+// The List of fields should Always be given at the last.
+// The TransactionDefinitionOptionsList will consist of all the Transaction Options
+TransactionOptions <- txnOptions:TransactionDefinitionOptionsList? SPACE? fields:FieldList?  {
+
+    transactionRequest := &structs.TransactionArguments{
+        Fields:     nil,
+        StartsWith: "",
+        EndsWith:   "",
+    }
+
+    if fields != nil {
+        transactionRequest.Fields = fields.([]string)
+    }
+     
+    if txnOptions != nil {
+        txnArgs := txnOptions.(*TxnArgs)
+        transactionRequest.StartsWith = txnArgs.startsWith
+        transactionRequest.EndsWith = txnArgs.endsWith
+    }
+
+    return transactionRequest, nil
+}
+
+// Parses one or more FieldNames separate by a space
+// Returns a slice containing all the FieldNames
+FieldList <- first:FieldName rest:(SPACE FieldName)* {
```

Comment:
There's already a `FieldNameList` that is comma separated. Since this has to be space separated, maybe rename this new rule something like `SpaceSeparatedFieldNameList`

Reply:
I have the name as you have suggested.